xmovie/core.py

```diff
@@ -358,12 +361,65 @@ def save_frames(self, odir, progress=False):
         elif ~tqdm_avail and progress:
             warnings.warn("Cant show progess bar at this point. Install tqdm")
 
-        for fi in frame_range:
-            fig, ax, pp = self.render_frame(fi)
-            frame_save(
-                fig, fi, odir=odir, frame_pattern=self.frame_pattern, dpi=self.dpi
+        for timestep in frame_range:
+            fig, ax, pp = self.render_single_frame(timestep)
+            save_single_frame(
+                fig, timestep, odir=odir, frame_pattern=self.frame_pattern, dpi=self.dpi
+            )
+
+    def save_frames_parallel(self, odir, parallel_compute_kwargs=dict()):
+        """
+        Saves all frames in parallel using dask.map_blocks.
+
+        Parameters
+        ----------
+        odir : path
+            path to output directory
+        parallel_compute_kwargs : dict
+            Keyword arguments to pass t dask's `compute()` function.
+
+        """
+        import numpy as np
+        import dask.array as darray
+
+        da = self.data
+        framedim = self.framedim
+
+        # Ensure that `da` has single chunks along `framedim`. Otherwise this might result in unexpected output.
+        framedim_chunks = da.chunks
+        if framedim_chunks is None:
+            raise ValueError(
+                f"Input data needs to be a dask array to save in parallel. Please chunk the input with single chunks along {framedim}."
+            )
+        framedim_chunks = framedim_chunks[da.dims.index(framedim)]
```

Comment:
minor: this looks cleaner?

```suggestion
        if da.chunks is None:
            raise ValueError(
                f"Input data needs to be a dask array to save in parallel. Please chunk the input with single chunks along {framedim}."
            )
        framedim_chunks = da.chunks[da.dims.index(framedim)]
```

Reply:
LGTM!